### PR TITLE
added CinemaHall-, Movie- and MovieSession- controllers

### PR DESCRIPTION
**Pull request by IgorBochkor**
added CinemaHall-, Movie- and MovieSession- controllers

### CODE REVIEW

**Review by antoniukivan — APPROVED** [review]

File: src/main/java/com/dev/cinema/controllers/CinemaHallController.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/cinema-halls")
+public class CinemaHallController {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controllers/CinemaHallController.java]: redundant empty line

File: src/main/java/com/dev/cinema/controllers/CinemaHallController.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                                CinemaHallMapper cinemaHallMapper) {
+        this.cinemaHallService = cinemaHallService;
+        this.cinemaHallMapper = cinemaHallMapper;
+    }
+
+    @PostMapping
+    public void create(@RequestBody CinemaHallRequestDto cinemaHallRequestDto) {
+        CinemaHall cinemaHall = new CinemaHall();
+        cinemaHall.setDescription(cinemaHallRequestDto.getDescription());
+        cinemaHall.setCapacity(cinemaHallRequestDto.getCapacity());
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controllers/CinemaHallController.java]: this should be done in the mapper

File: src/main/java/com/dev/cinema/controllers/MovieController.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    public MovieController(MovieMapper movieMapper, MovieService movieService) {
+        this.movieMapper = movieMapper;
+        this.movieService = movieService;
+    }
+
+    @PostMapping
+    public void create(@RequestBody MovieRequestDto movieRequestDto) {
+        Movie movie = new Movie();
+        movie.setDescription(movieRequestDto.getDescription());
+        movie.setTitle(movieRequestDto.getTitle());
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controllers/MovieController.java]: same

File: src/main/java/com/dev/cinema/controllers/MovieSessionController.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    public void update(@PathVariable Long id,
+                       @RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        MovieSession movieSession
+                = movieSessionMapper.mapRequestDtoToMovieSession(movieSessionRequestDto);
+        movieSession.setId(id);
+        movieSessionService.update(movieSession);
+    }
+
+    @DeleteMapping("/{id}")
+    public void del(@PathVariable Long id) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controllers/MovieSessionController.java]:
```suggestion
    public void delete(@PathVariable Long id) {
```

File: src/main/java/com/dev/cinema/dao/MovieSessionDao.java [hunk 5]
```diff
@@ -5,7 +5,12 @@
 import java.util.List;
 
 public interface MovieSessionDao {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dao/MovieSessionDao.java]: redundant empty line

File: src/main/java/com/dev/cinema/model/dto/MovieSessionRequestDto.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    public String getShowTime() {
+        return showTime;
+    }
+
+    public void setShowTime(String showTime) {
+        this.showTime = showTime;
+    }
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/dto/MovieSessionRequestDto.java]: redundant empty line

File: src/main/java/com/dev/cinema/service/CinemaHallService.java [hunk 7]
```diff
@@ -7,4 +7,6 @@
     CinemaHall add(CinemaHall cinemaHall);
 
     List<CinemaHall> getAll();
+
+    CinemaHall getCinemaHall(Long id);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/CinemaHallService.java]:
```suggestion
    CinemaHall get(Long id);
```

File: src/main/java/com/dev/cinema/service/MovieService.java [hunk 8]
```diff
@@ -7,4 +7,6 @@
     Movie add(Movie movie);
 
     List<Movie> getAll();
+
+    Movie getMovie(Long id);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/MovieService.java]:
```suggestion
    Movie get(Long id);
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/controllers/CinemaHallController.java (hunk 2 above)
Comment (IgorBochkor) [reply src/main/java/com/dev/cinema/controllers/CinemaHallController.java]: fixed

File: src/main/java/com/dev/cinema/controllers/MovieController.java (hunk 3 above)
Comment (IgorBochkor) [reply src/main/java/com/dev/cinema/controllers/MovieController.java]: fixed

File: src/main/java/com/dev/cinema/controllers/MovieSessionController.java (hunk 4 above)
Comment (IgorBochkor) [reply src/main/java/com/dev/cinema/controllers/MovieSessionController.java]: fixed

File: src/main/java/com/dev/cinema/service/CinemaHallService.java (hunk 7 above)
Comment (IgorBochkor) [reply src/main/java/com/dev/cinema/service/CinemaHallService.java]: fixed

File: src/main/java/com/dev/cinema/service/MovieService.java (hunk 8 above)
Comment (IgorBochkor) [reply src/main/java/com/dev/cinema/service/MovieService.java]: fixed

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**